src/validation.cairo

```diff
@@ -319,16 +338,17 @@ fn compute_block_reward(block_height: u32) -> u64 {
 #[cfg(test)]
 mod tests {
     use raito::state::{Header, Transaction, TxIn, TxOut, OutPoint};
-    use raito::utils::Hash;
+    use raito::utils::{Hash, HashTrait};
     use raito::test_utils::from_hex;
+    use raito::merkle_tree::merkle_root;
     use super::{
         validate_timestamp, validate_proof_of_work, compute_block_reward, compute_total_work,
         compute_work_from_target, shr, shl, Block, ChainState, UtreexoState, next_prev_timestamps,
-        TransactionValidatorImpl, validate_coinbase, bits_to_target, target_to_bits
+        TransactionValidatorImpl, validate_coinbase, bits_to_target, target_to_bits, block_hash
     };
 
     #[test]
-    fn test_validate_timestamp() {
+    fn test_block_hash() {
         let mut chain_state = ChainState {
             block_height: Option::Some(1),
             total_work: 1,
```

Comment:
Just put arbitrary best block hash here, it does not matter for the unit test

Reply:
I dont get this

Reply:
You need to create a unit test to make sure your method returns an expected result.

1. Go to https://learnmeabitcoin.com/technical/block/hash/
2. Press "Random example"
3. Save `Block Hash (Natural Byte Order)` - this is your expected block hash
4. Copy "Block Hash (Reverse Byte Order)"
5. Go to https://blastapi.io/public-api/bitcoin
6. Select method "getblockheader"
7. Paste the copied block hash
8. Copy all the necessary fields from the response

The step with learnmeabitcoin is needed to troubleshoot your encoding issues (e.g. identify that you serialized the data in an incorrect order or wrong endianness)